### ISSUE TITLE
[ProvenanceAnalysis] Assertion failed: (relatedCheck(B, A) == Result && "relatedCheck result depending on order of parameters!")

### ISSUE THREAD

**Issue by dcci** [llvm/llvm-project#58864]
```console
% ./clang reduced.m -fobjc-arc -Os -c
```

```console
Assertion failed: (relatedCheck(B, A) == Result && "relatedCheck result depending on order of parameters!"), function related, file ProvenanceAnalysis.cpp, line 187.
```

Testcase:
```mm
# 48 "/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/arm/_types.h" 3 4
_OSSwapInt16(
 )
{
} filesec_property_t;
typedef long NSInteger;
typedef unsigned long NSUInteger;
@protocol NSObject
@end
@interface NSObject <NSObject> {
}
+ (instancetype)alloc __attribute__((availability(swift, unavailable, message="use object initializers instead")));
@end
#pragma clang assume_nonnull begin
@protocol NSCopying
@end
@protocol NSMutableCopying
@end
@protocol NSCoding
@end
@protocol NSSecureCoding <NSCoding>
@end
@protocol NSFastEnumeration
@end
@interface NSString : NSObject <NSCopying, NSMutableCopying, NSSecureCoding>
typedef enum __attribute__((flag_enum,enum_extensibility(open))) NSStringEnumerationOptions : NSUInteger NSStringEnumerationOptions; enum NSStringEnumerationOptions : NSUInteger {
    NSStringEnumerationByLines = 0,
};
- (nullable instancetype)initWithUTF8String:(const char *)nullTerminatedCString __attribute__ ((format_arg(1)));
@end
@interface NSDictionary<__covariant KeyType, __covariant ObjectType> : NSObject <NSCopying, NSMutableCopying, NSSecureCoding, NSFastEnumeration>
+ (instancetype)dictionaryWithObjects:(const ObjectType _Nonnull [_Nullable])objects forKeys:(const KeyType <NSCopying> _Nonnull [_Nullable])keys count:(NSUInteger)cnt;
@end
#pragma clang assume_nonnull end
typedef NSString *NSErrorDomain;
#pragma clang assume_nonnull begin
extern NSErrorDomain const NSPOSIXErrorDomain;
typedef NSString *NSErrorUserInfoKey;
extern NSErrorUserInfoKey const NSLocalizedDescriptionKey;
extern NSErrorUserInfoKey const NSLocalizedFailureReasonErrorKey;
@interface NSError : NSObject <NSCopying, NSSecureCoding> {
}
+ (instancetype)errorWithDomain:(NSErrorDomain)domain code:(NSInteger)code userInfo:(nullable NSDictionary<NSErrorUserInfoKey, id> *)dict;
@end
struct __attribute__((availability(ios,unavailable))) __attribute__((availability(watchos,unavailable))) __attribute__((availability(tvos,unavailable))) __attribute__((availability(macCatalyst,unavailable))) SecKeychainAttribute
{
};
#pragma clang assume_nonnull end
#pragma clang assume_nonnull begin
enum __attribute__((enum_extensibility(open))) : NSInteger {
    NSNoSubelement = 4
};
#pragma clang assume_nonnull end
@implementation _DVTErrorRecoveryHandler
+ (NSError *)dvt_errorWithPOSIXErrorCode:(NSInteger)errorCode messageFormat:(NSString *)msgFormat, ...
{
    NSString * rsnString = [[NSString alloc] initWithUTF8String:strerror(errorCode)];
    NSString * keys[2] = { NSLocalizedDescriptionKey, NSLocalizedFailureReasonErrorKey };
    NSString * objs[2] = { rsnString, rsnString };
    NSDictionary * errorDict = [NSDictionary dictionaryWithObjects:objs forKeys:keys count:2];
    return [NSError errorWithDomain:NSPOSIXErrorDomain code:errorCode userInfo:errorDict];
}
@end
```

**Comment by ahatanak** [comment]
Both A and B are load instructions and both `AIsIdentified` and `BIsIdentified` are true. `relatedCheck(B, A)` and `relatedCheck(A, B)` disagree because `IsStoredObjCPointer(A)` and `IsStoredObjCPointer(B)` return different values.

**Comment by ahatanaka** [comment]
Fixed in 295861514e0d1e48df2918b630dd692ac27ee0de
